index.js

```diff
@@ -36,12 +35,32 @@ var point = require('turf-point');
 module.exports = function (line, units) {
   var coords;
   if(line.type === 'Feature') coords = line.geometry.coordinates;
-  else if(line.type === 'LineString') coords = line.geometry.coordinates;
+  else if(line.type === 'LineString') coords = line.coordinates;
   else throw new Error('input must be a LineString Feature or Geometry');
 
   var travelled = 0;
-  for(var i = 0; i < coords.length - 1; i++) {
-    travelled += distance(point(coords[i]), point(coords[i+1]), units);
+  var prevCoords = {
+    type: 'Feature',
+    properties: {},
+    geometry: {
+      type: 'Point',
+      coordinates: coords[0]
+    }
+  };
+  var curCoords = {
+      type: 'Feature',
+      properties: {},
+      geometry: {
+        type: 'Point'
+      }
+    };
```

Comment:
we can cut some lines here with `var prevCoords = point(coords[0])`, same for `curCoords`.